0032-peering-clearing-settlement/0032-peering-clearing-settlement.md

```diff
@@ -0,0 +1,160 @@
+
+In financial accounting terms this could be viewed as a revenue/expense account where credits to this account are balanced by a debit to a corresponding "Accounts Receivable" asset account and debits to this account are balanced by a credit to a corresponding "Accounts Payable" liability account.
+
+> **Example : Alice and Bob**
+
+> Assuming they start with a balance of 0 and the following sequence of events occurs:
+
+> 1. Alice passes an ILP Prepare to Bob with an amount of 6.
+> 1. Bob replies, before the request expires, with a valid ILP Fulfill.
+> 1. Bob passes an ILP Prepare to Alice with an amount of 2.
+> 1. Alice is unable to route the packet and replies with an ILP Reject.
+> 1. Bob passes an ILP Prepare to Alice with an amount of 20.
+> 1. Alice replies, before the request expires, with a valid ILP Fulfill.
+
+> The balance changes will be as follows:
+
+> - After Step 2:
+>   - Alice's balance is -6 satoshis
+>   - Bob's balance is 6 satoshis
+> - After Step 4 there are no changes to the balance as the packet was not delivered.
+> - After Step 6:
+>   - Alice's balance is 14 satoshis
+>   - Bob's balance is -14 satoshis
+
+# Clearing
+
+The traditional definition of clearing a payment, is the process by which the two parties establish their net positions and reconcile any differences.
+
+In the Interledger case this would involve the two peers exchanging data on what their current balance is and, if there is a discrepency, resolving this.
+
+Notably, clearing is not part of the Interledger standard, as the means by which two peers perform these operations is a bilateral concern (i.e. has no impact on others in the network).
+
+Current implementations of ILP do not have a standardized clearing process, as most peers settle very frequently using networks on which this is possible such as distributed ledgers which support payment channels, reducing the risk of discrepencies developing in their account balances.
+
+However, there is an edge case where the peer's balances may be out of sync demonstrated in this example.
+
+> **Example : Alice and Bob**
+
+> Assuming they start with a balance of 0 and the following sequence of events occurs:
+
+> 1. Bob passes an ILP Prepare to Alice with an amount of 20.
+> 1. Alice replies with a valid ILP Fulfill and believes she has done so before the request expires, however Bob considers the request to have expired and has already rejected the payment to his downstream peer.
+
+> The balance changes will be as follows:
+
+> - After Step 2:
+>   - Alice's balance is 20 satoshis
+>   - Bob's balance is 0 satoshis
+
+> Clearly, Bob and Alice need to reconcile this situation. The specific mechanism they use and recourse they take to resolve this difference is outside the scope of the Interledger protocol. 
+
+It is important to note that this is an extremely unlikely scenario as telescoping expiry timeouts between peers should reduce the risk of this to almost zero.
```

Comment:
It may happen but nodes control how much time they subtract from the expiry time before forwarding it (worth explaining the telescoping bit and emphasizing that it's under their control)